### ISSUE TITLE
Cannot read property `map` of `undefined`

### ISSUE THREAD

**Issue by ehmicky** [Munter/netlify-plugin-subfont#39]
The following error happens in some builds:

```
Failed during font subsetting
Cannot read property 'map' of undefined 
    /opt/build/repo/node_modules/font-snapper/lib/fontSnapper.js:177:6 snapToAvailableFontProperties
    /opt/build/repo/node_modules/subfont/lib/subsetFonts.js:119:41 families.map.family
    /opt/build/repo/node_modules/subfont/lib/subsetFonts.js:118:21 _.flatMapDeep.textAndProps
    /opt/build/repo/node_modules/lodash/lodash.js:639:23 arrayMap
    /opt/build/repo/node_modules/lodash/lodash.js:9554:14 map
    /opt/build/repo/node_modules/lodash/lodash.js:9281:26 Function.flatMapDeep
    /opt/build/repo/node_modules/subfont/lib/subsetFonts.js:102:26 groupTextsByFontFamilyProps
    /opt/build/repo/node_modules/subfont/lib/subsetFonts.js:698:23 subsetFonts
```

Do you know what the problem might be?

**Comment by Munter** [comment]
Not right off the bat. But I don know that the line number seems wrong: https://github.com/assetgraph/font-snapper/blob/master/lib/fontSnapper.js#L177

But there is certainly a lot of `.map` calls in font-snapper. Luckily the library is pretty well covered with tests, so I might be able to have a look at the callstack and imagine the case

**Comment by Munter** [comment]
It looks like we are getting a value in in the `@font-face` declaration that is not one of `normal`, `italic`, `oblique`.  This is the relevant piece of code:

```js
  let fontStyle = propsToSnap['font-style'].toLowerCase();
  // Don't break on oblique <angle>
  if (/^oblique\b/.test(fontStyle)) {
    fontStyle = 'oblique';
  }

  // Find the best font-style
  const styleMatches = styleLookupOrder[fontStyle]
    .map(style => // This is what throws. styleLookupOrder[fontStyle] is undefined
      stretchMatches.filter(
        stretchMatch => stretchMatch['font-style'].toLowerCase() === style
      )
    )
    .find(list => list.length > 0);
```

So either we are getting invalid inputs, or our handling of oblique angles is wrong. @paapandreou we probably need to think about what to do with possibly invalid input here. Throw? Match nothing? I'm pretty sure the browser will snap to something.

It would really be great if we could know what page this is and find the offending font-face declaration

**Comment by Munter** [comment]
Released in v5.0.1
